README.md

```diff
@@ -228,7 +228,7 @@ to a command update to the device.
 
 ![General ](https://raw.githubusercontent.com/telefonicaid/iotagent-node-lib/master/img/iotAgentLib.png "Architecture Overview")
 
-### <a name="TimeInstant"/>The ´TimeInstant´ element
+### The ´TimeInstant´ element
```

Comment:
Looking in the Orion documentation it seems that `<a name=...>` works ok. For intance, in https://raw.githubusercontent.com/telefonicaid/fiware-orion/master/doc/manuals/admin/metrics_api.md we have:

```
# <a name="top"></a>Metrics API
```

and in the RTD rendering of that document (at https://fiware-orion.readthedocs.io/en/master/admin/metrics_api/index.html) it looks fine:

![imagen](https://user-images.githubusercontent.com/1534240/41222231-9b12d956-6d66-11e8-9ca4-9d8ffa2a6f33.png)

and the links with "Top" also seem to work fine.

So, could you elaborate on the problem with anchors in this repo, please? Maybe is a matter of chaning `<a name=.../>` by the one seems to work for Orion repo, i.e. `<a name=...></a> (althouth from a HTML point of view is the same, maybe something in RTD causes problems with the former).

Reply:
I agree the HTML rendering **should** be the same, except that it isn't. [sigh]

Go to https://iotagent-node-lib.readthedocs.io/en/latest/usermanual/index.html

Hover over the line:
>The library provides a mechanism for the periodic reporting of stats related to the library's work.
>In order to activate the use of the periodic stats, it must be configured in the config file, as 
>described in the Configuration section.
>

The **entire paragraph** appears to be a link. This is due to readthedocs mis-interpreting:

```html
### <a name="statsregistry"/> Stats Registry
```

in the line before as an open `<a href>` anchor tag

Of course the rendering of GFM on a Markdown file on **GitHub** itself is fine - it's readthedocs that can't handle it.

The rendering of the sidebar nav is also completely screwed up as well.

I would argue that in general  adding HTML `<a href>`  tags into Markdown files (although technically acceptable within the specifications most Markdown flavors) are:

* A Totally unnecessary hack (since Markdown has its own linking mechanism)
* More difficult to maintain than generating markdown style links
* Make the Raw Markdown file more difficult to read.

The main acceptable usage for  `<a href>` tags is for markdown files written in extended character sets (e.g. CJK) where the markdown ascii character linking just doesn't work. In general titles for files written in English won't have these issues.

Reply:
I mostly agree, but not sure if somebody is using these anchors (e.g. in direct links to documentation in other places)... should that people get fails after the "cleanup"?

Maybe we can use this PR to change `<a name="..." />` to `<a name="..."></a> (as in Orion RTD rendering it seems to work) and see how it renders in RTD. If after doing that it is still having problem, we can remove `<a>` usages completely.

What do you think?

Reply:
My opinion, as stated above - `<a name="."/>` should be **deleted** - easier to read, easier to maintain.

Since this document wasn't even on readthedocs until two weeks ago, I doubt there are any **external** deep links. Even then, usually it would make more sense to deep link to the head of each file if a deep-link is necessary e.g. https://iotagent-node-lib.readthedocs.io/en/latest/usermanual/ 

So the only likely broken links will be found within the document itself. The ToC has been regenerated so those links should work, so it can be reduced to internal links within the body of the document. Personally if I come across a dead link in the body of the document I'd probably click on the sidebar navigation (which removing `<a name="."/>`  tag fixes in readthedocs anyway)

Besides which, a failed deep link will also display the head of the document e.g.: https://iotagent-node-lib.readthedocs.io/en/latest/usermanual/#x - at worst this will be an `<h1>` rather than an `<h3>` 

Finally,  if we leave `<a name="."/>`  tags in there you are only encouraging someone else to start using them.

I say **delete** :smiley:

Reply:
The document wasn't even on readthedocs until two weeks ago but it has been time along in markdown. Thus, people using links like

https://github.com/jason-fox/iotagent-node-lib/blob/master/README.md#TimeInstant

will have them broken after merging PR.

However, I think is a good "hygienic" change the two ways of linking. So people that get the above link broken (not so broken, as at the end, they will land at the top of README.md) could change to the right one according to markdown internal way of linking, i.e:

https://github.com/jason-fox/iotagent-node-lib/blob/master/README.md#the-timeinstant-element

Thus, NTC :)